### PR TITLE
Refactor translation module + translation using mitochondrial genetic code

### PR DESCRIPTION
**Pull request by iskandr**
We have been, thus far, using the 'standard' genetic code to translate all cDNA sequences, even if they're on the mitochondrial genome! 

This PR splits up the `translation` module into smaller parts, adds a `genetic_code` module with two codon tables (not going to add the more obscure bacterial/archaean codon tables) and uses the variant's contig to switch between cytosolic vs. mitochondrial translation.

Got rid of dependency on skbio, fixes https://github.com/hammerlab/isovar/issues/29

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/isovar/60)

### CODE REVIEW

File: isovar/translation.py [hunk 1]
```diff
@@ -102,185 +59,132 @@
...
+        if protein_sequence_length and len(variant_amino_acids) > protein_sequence_length:
+            if protein_sequence_length <= variant_aa_interval_start:
+                logger.warn(
+                    ("Truncating amino acid sequence %s from variant sequence %s "
+                     "to only %d elements loses all variant residues"),
+                    variant_amino_acids,
+                    variant_sequence,
+                    protein_sequence_length)
+                return None
+            # if the protein is too long then short it, which implies we're no longer
```
Comment (ihodes) [inline isovar/translation.py]: shorten*

File: isovar/genetic_code.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            if stop_codon in self.codon_table:
+                if self.codon_table[stop_codon] != "*":
+                    raise ValueError(
+                        "Stop codon '%s' must translate to '*'" % (
+                            stop_codon,))
+            else:
+                self.codon_table[stop_codon] = "*"
+        for start_codon in self.start_codons:
+            if start_codon not in self.codon_table:
+                raise ValueError(
```
Comment (ihodes) [inline isovar/genetic_code.py]: Why can't you add start codons like you do stop codons?

Comment (ihodes) [inline isovar/genetic_code.py]: (And what sentinel value do you use for start codons?)

File: isovar/genetic_code.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                    "Start codon '%s' missing from codon table" % (
+                        start_codon,))
+        for codon, amino_acid in self.codon_table.items():
+            if amino_acid == "*" and codon not in self.stop_codons:
+                raise ValueError(
+                    "Non-stop codon '%s' can't translate to '*'" % (
```
Comment (ihodes) [inline isovar/genetic_code.py]: Could something like "Codon '%s' not found in stop_codons, but codon_table indicates that it should be" be a little clearer?

File: isovar/genetic_code.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        cdna_sequence : str
+            cDNA sequence which is expected to start and end on complete codons.
+
+        first_codon_is_start : bool
+            Is the first codon of the sequence a start codon?
+        """
+        if not isinstance(cdna_sequence, str):
+            cdna_sequence_from_first_codon = str(cdna_sequence)
+        n = len(cdna_sequence_from_first_codon)
+        if len(cdna_sequence_from_first_codon) < 3:
```
Comment (ihodes) [inline isovar/genetic_code.py]: Can use `n` here (or maybe rename to `length_of_cdna_seq`?)

File: isovar/genetic_code.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+            return ''
```
Comment (ihodes) [inline isovar/genetic_code.py]: Exception might make more sense, since we're breaking the function contract?

Comment (ihodes) [inline isovar/genetic_code.py]: A bit pedantic though. Don't really care, esp in interest of expediency here.

File: isovar/genetic_code.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        codon_table = self.codon_table
+        if first_codon_is_start and cdna_sequence[:3] in self.start_codons:
+            amino_acid_list = ['M']
+            start_index = 3
+        else:
+            start_index = 0
+            amino_acid_list = []
+
+        # trim to multiple of 3 length
+        end_idx = 3 * (n // 3)
```
Comment (ihodes) [inline isovar/genetic_code.py]: Another place to error (i.e. if seq is not a multiple of three, something is wrong)

File: isovar/genetic_code.py (hunk 2 above)
Comment (iskandr) [inline isovar/genetic_code.py]: Start codons, when they're actually being used to initiate translation, almost always translate as methionine (except for a few leucine exceptions that I don't yet know how to deal with), but within the protein might translate as something else (e.g. CTG = leucine, GTG = valine).

File: isovar/common.py [hunk 7]
```diff
@@ -16,24 +16,59 @@
...
 
 def reverse_complement_dna(seq):
     return complement_dna(seq)[::-1]
 
+
+# TODO: move all the DNA manipulating code out to its own library
+#
+# map each degenerate DNA letter code to the set of core nucleotides
+# it represents
+degenerate_dna_nucleotide_dict = {
```
Comment (ihodes) [inline isovar/common.py]: What is this used for? And references for the mapping?

File: isovar/genetic_code.py (hunk 3 above)
Comment (iskandr) [inline isovar/genetic_code.py]: 👍

File: isovar/genetic_code.py (hunk 4 above)
Comment (iskandr) [inline isovar/genetic_code.py]: 👍

File: isovar/genetic_code.py (hunk 6 above)
Comment (iskandr) [inline isovar/genetic_code.py]: I don't think it's an error really. If an mRNA is missing its stop codon then the ribosome eventually falls off -- regardless of the sequence length. And if there's a stop codon before the last incomplete codon then the ribosome falls off there.

File: isovar/common.py (hunk 7 above)
Comment (iskandr) [inline isovar/common.py]: Moved to https://github.com/iskandr/bioseq for now

**Pull request merged**